content/articles/building-a-password-strength-classifier-model-using-machine-learning/index.md

```diff
@@ -0,0 +1,351 @@
+
+From the image above, our dataset has one missing value. Let's remove the missing value.
+
+#### Removing missing values
+
+We use the following code:
+
+```python
+data.dropna(inplace=True)
+```
+To check if the missing value is removed, run this code:
+
+```python
+data.isnull().sum()
+```
+The output is shown below:
+
+![Removing missing values](/engineering-education/building-a-password-strength-classifier-model-using-machine-learning/removing-missing-values.jpg)
+
+After removing the missing values, we will convert our dataset into an array.
+
+### Convert dataset into an array
+
+An array is much easier to work with, we will use the NumPy Python library. To import NumPy use this code:
+
+```python
+pswd = np.array(pswd_data)
+```
+To see our the array, run this code:
+
+```python
+pswd
+```
+The output is shown below:
+
+![Removing missing values](/engineering-education/building-a-password-strength-classifier-model-using-machine-learning/dataset-array.jpg)
+
+The next step is to shuffle our dataset randomly. Shuffling the dataset will make the dataset more robust. This will prevent the model from memorizing the dataset and ensure the model learns from the dataset. Using a shuffled dataset the model will understand patterns and relationships within our dataset.
+
+### Dataset shuffling
+
+To shuffle our dataset, we will import the `random` Python package.
+
+```python
+import random
+```
+We can now randomly shuffle the dataset using the following code:
+
+```python
+random.shuffle(pswd)
+```
+The next step is to add features and labels to our dataset.
+
+### Adding features and labels
+
+In machine learning, features are all the unique independent variables in our dataset that are used as the model's input. Here, our features are located in the `password` column. The `password` column holds all the text passwords that will train our model.
+
+Labels are variables in our dataset that are used as output for the model. Here, our labels are located in the `strength` column. The `strength` column has three values, 0 to 2.
+
+We will add the features and labels using the following code:
+
+```python
+ylabels  = [s[1] for s in pswd]
+allpasswords = [s[0] for s in pswd]
+```
+In the code above we save our labels in the `ylabels` variable. The code loops through the columns and selects the last column(1) as the labels. We also save our features in the `allpasswords` column. The code loops through the columns and selects the column(0) as the features.
+
+To check the number of passwords and labels in our dataset, run this code:
+
+- Number of labels
+
+```python
+len(ylabels)
+```
+The output is shown below:
+
+```bash
+669639
+```
+- Number of passwords
+
+```python
+len(allpasswords)
+```
+The output is shown below:
+
+```bash
+669639
+```
+From the image above this dataset is big and has many data samples, we can now use this dataset to build our model.
+
```

Comment:
kindly add the image you are talking about

Reply:
Hello @M0nicah, that was an oversight. It's supposed to be from the `output` above not `image` above. I have made that change .
Thanks